src/Cake.AzureDevOps/Pipelines/AzureDevOpsBuild.cs

```diff
@@ -401,6 +405,63 @@ public IEnumerable<AzureDevOpsBuildArtifact> GetArtifacts()
             }
         }
 
+        /// <summary>
+        /// Gets the test run ID's of a build.
+        /// </summary>
+        /// <returns>A list of RunIDs as int.</returns>
+        public IEnumerable<AzureDevOpsTestRun> GetTestRuns()
+        {
+            if (!this.ValidateBuild())
+            {
+                return new List<AzureDevOpsTestRun>();
+            }
+
+            using (var testClient = this.testClientFactory.CreateTestManagementClient(this.CollectionUrl, this.credentials.ToVssCredentials()))
+            {
+                return
+                    testClient
+                        .GetTestResultDetailsForBuildAsync(this.ProjectId, this.build.Id)
+                        .ConfigureAwait(false)
+                        .GetAwaiter()
+                        .GetResult()
+                        .ResultsForGroup
+                            .SelectMany(testResultGroup => testResultGroup.Results).ToList()
+                            .GroupBy(testResult => testResult.TestRun.Id)
+                            .Select(runGroup => int.Parse(runGroup.Key))
+                            .Select(runId =>
+                                new AzureDevOpsTestRun
+                                {
+                                    RunId = runId,
+                                    TestResults = this.GetTestResults(testClient, runId),
+                                });
+            }
+        }
+
+        /// <summary>
+        /// Gets the test runs of this build.
+        /// </summary>
+        /// <param name="testClient">Instance of a <see cref="TestManagementHttpClient"/> class.</param>
+        /// <param name="runId">Id of the test run.</param>
+        /// <returns>The test results for a build or an empty list if no build could be found and
+        /// <see cref="AzureDevOpsBuildSettings.ThrowExceptionIfBuildCouldNotBeFound"/> is set to <c>false</c>.</returns>
+        /// <exception cref="AzureDevOpsBuildNotFoundException">If build could not be found and
+        /// <see cref="AzureDevOpsBuildSettings.ThrowExceptionIfBuildCouldNotBeFound"/> is set to <c>true</c>.</exception>
+        private IEnumerable<AzureDevOpsTestResult> GetTestResults(TestManagementHttpClient testClient, int runId)
```

Comment:
Make this method async and await the reading of the test results so that it can make multiple calls to read the test results in parallel.